pkg/blockstorage/azure/azuredisk.go

```diff
@@ -162,8 +162,9 @@ func (s *adStorage) SnapshotCopy(ctx context.Context, from blockstorage.Snapshot
 	if err != nil {
 		return nil, err
 	}
-	blob := container.GetBlobReference(copyBlobName)
-	defer deleteBlob(blob)
+	blobName := fmt.Sprintf(copyBlobName, name)
```

Comment:
Looks good iff name is composed of valid characters for a blob name. Did you check?

Reply:
Yes, the blockstorage_test passes for this change.